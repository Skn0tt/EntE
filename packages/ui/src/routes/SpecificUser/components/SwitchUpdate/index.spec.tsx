import * as React from "react";
import { shallow } from "enzyme";
import { SwitchUpdate } from "./";
import * as sinon from "sinon";

describe("SwitchUpdate", () => {
  const onChange = sinon.spy();
  const comp = shallow(
    <SwitchUpdate title="Switch" onChange={onChange} value={false} />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });

  describe("when clicking switch", () => {
    comp
      .find(".updateSwitch")
      .simulate("change", { target: { checked: true } });

    it("calls `onChange`", () => {
      expect(onChange.calledWithExactly(true)).toBe(true);
    });
  });
});
