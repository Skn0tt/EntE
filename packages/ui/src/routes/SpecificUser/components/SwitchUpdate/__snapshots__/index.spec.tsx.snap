// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchUpdate renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onChange={[Function]}
    title="Switch"
    value={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(GridWrapper)
          item={true}
        >
          <WithStyles(Typography)
            variant="title"
          >
            Switch
          </WithStyles(Typography)>
        </WithStyles(GridWrapper)>,
        <WithStyles(GridWrapper)
          item={true}
        >
          <WithStyles(Switch)
            checked={false}
            className="updateSwitch"
            onChange={[Function]}
          />
        </WithStyles(GridWrapper)>,
      ],
      "container": true,
      "direction": "column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            variant="title"
          >
            Switch
          </WithStyles(Typography)>,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Switch",
            "variant": "title",
          },
          "ref": null,
          "rendered": "Switch",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Switch)
            checked={false}
            className="updateSwitch"
            onChange={[Function]}
          />,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "className": "updateSwitch",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(GridWrapper)
            item={true}
          >
            <WithStyles(Typography)
              variant="title"
            >
              Switch
            </WithStyles(Typography)>
          </WithStyles(GridWrapper)>,
          <WithStyles(GridWrapper)
            item={true}
          >
            <WithStyles(Switch)
              checked={false}
              className="updateSwitch"
              onChange={[Function]}
            />
          </WithStyles(GridWrapper)>,
        ],
        "container": true,
        "direction": "column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              variant="title"
            >
              Switch
            </WithStyles(Typography)>,
            "item": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Switch",
              "variant": "title",
            },
            "ref": null,
            "rendered": "Switch",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Switch)
              checked={false}
              className="updateSwitch"
              onChange={[Function]}
            />,
            "item": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "className": "updateSwitch",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
